src/AstroImages.jl

```diff
@@ -85,9 +85,21 @@ for n in (8, 16, 32, 64)
     end
 end
 
+mutable struct properties{T <: AbstractArray}
+    rgb_image::T
+    function properties{T}(;kvs...) where T
+        obj = new{T}()
+        for (k,v) in kvs
+            setproperty!(obj, k, v)
+        end
+        return obj
+    end
+end
+
 struct AstroImage{T<:Real,C<:Color, N}
     data::NTuple{N, Matrix{T}}
     wcs::NTuple{N, WCSTransform}
+    property::properties
```

Comment:
This is an abstract type, not a concrete one, as `properties` is parametrised.

Reply:
What are you suggesting here? Do the syntax needs to be changed in any way here?

Reply:
using the proper paramter